src/java_tools/junitrunner/javatests/com/google/testing/coverage/JacocoLCOVFormatterUninstrumentedTest.java

```
@@ -0,0 +1,176 @@
+public class JacocoLCOVFormatterUninstrumentedTest {
+
+  private StringWriter writer;
+  private IBundleCoverage mockBundle;
+
+  private static IClassCoverage mockIClassCoverage(String className, String packageName, String sourceFileName) {
+    IClassCoverage mocked = mock(IClassCoverage.class);
+    when(mocked.getName()).thenReturn(className);
+    when(mocked.getPackageName()).thenReturn(packageName);
+    when(mocked.getSourceFileName()).thenReturn(sourceFileName);
+    return mocked;
+  }
+
+  private Description createSuiteDescription(String name) {
+    Description suite = Description.createSuiteDescription(name);
+    suite.addChild(Description.createTestDescription(Object.class, "child"));
+    return suite;
+  }
+
+  @Before
+  public void setupTest() {
+    // Initialize writer for storing coverage report outputs
+    writer = new StringWriter();
+    // Initialize mock Jacoco bundle containing the mock coverage
+    // Classes
+    List<IClassCoverage> mockClassCoverages = Arrays.asList(
+      mockIClassCoverage("Foo", "com/example", "Foo.java")
+    );
+    // Package
+    IPackageCoverage mockPackageCoverage = mock(IPackageCoverage.class);
+    when(mockPackageCoverage.getClasses()).thenReturn(mockClassCoverages);
+    // Bundle
+    mockBundle = mock(IBundleCoverage.class);
+    when(mockBundle.getPackages()).thenReturn(Arrays.asList(mockPackageCoverage));
+  }
+
+  @Test
+  public void testVisitBundleWithSimpleUnixPath() {
+    // Paths
+    ImmutableSet<String> execPaths = ImmutableSet.of(
+      "/parent/dir/com/example/Foo.java"
+    );
+    JacocoLCOVFormatter formatter = new JacocoLCOVFormatter(execPaths);
+    IReportVisitor visitor = formatter.createVisitor(new PrintWriter(writer), new TreeMap<String, BranchCoverageDetail>());
+    try {
+      visitor.visitBundle(mockBundle, mock(ISourceFileLocator.class));
+      visitor.visitEnd();
+    } catch (IOException e) {
+      Assert.fail(Throwables.getStackTraceAsString(e));
```

Why catch the IOException to then just "Assert.fail"? Just let the IOException bubble up and terminate the test.

---

Makes sense, I've updated the tests accordingly.